### PR TITLE
ODATA-1005 - open - Make sure we have capabilities for all new 4.01 functionality

### PR DESCRIPTION
**Pull request by ralfhandl**

### CODE REVIEW

File: vocabularies/Org.OData.Capabilities.V1.xml [hunk 1]
```diff
@@ -327,13 +328,20 @@ supported:
         <Annotation Term="Core.Description" String="Supports $skip" />
       </Term>
 
+      <Term Name="ComputeSupported" Type="Core.Tag" Nullable="false" DefaultValue="true" AppliesTo="EntitySet">
```
Comment (mikepizzo) [inline vocabularies/Org.OData.Capabilities.V1.xml]: shouldn't all of these be nullable?

File: vocabularies/Org.OData.Capabilities.V1.xml [hunk 2]
```diff
@@ -558,6 +571,18 @@ supported:
           String="Supports [key-as-segment convention](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part2-url-conventions.html#sec_KeyasSegmentConvention) for addressing entities within a collection" />
       </Term>
 
+      <Term Name="RelaxedSystemQueryOptionNamesSupported" Type="Core.Tag" Nullable="false" DefaultValue="true"
```
Comment (mikepizzo) [inline vocabularies/Org.OData.Capabilities.V1.xml]: Do we need this? I would just say it's required for 4.01, but that clients should use canonical formats for interoperability. I can't imagine a client querying this and then adjusting behavior based on the value.

File: vocabularies/Org.OData.Capabilities.V1.xml [hunk 3]
```diff
@@ -650,6 +679,12 @@ The absence of `RestrictedProperties` denotes all properties are accessible usin
         <Property Name="Updatable" Type="Edm.Boolean" Nullable="false" DefaultValue="true">
           <Annotation Term="Core.Description" String="Entities can be updated" />
         </Property>
+        <Property Name="Upsertable" Type="Edm.Boolean" Nullable="false" DefaultValue="false">
+          <Annotation Term="Core.Description" String="Entities can be upserted" />
+        </Property>
+        <Property Name="CollectionUpdate" Type="Capabilities.CollectionUpdateType" Nullable="false">
+          <Annotation Term="Core.Description" String="Entities can be upserted" />
```
Comment (mikepizzo) [inline vocabularies/Org.OData.Capabilities.V1.xml]: Is this the right description?

File: vocabularies/Org.OData.Capabilities.V1.xml [hunk 4]
```diff
@@ -650,6 +679,12 @@ The absence of `RestrictedProperties` denotes all properties are accessible usin
         <Property Name="Updatable" Type="Edm.Boolean" Nullable="false" DefaultValue="true">
           <Annotation Term="Core.Description" String="Entities can be updated" />
         </Property>
+        <Property Name="Upsertable" Type="Edm.Boolean" Nullable="false" DefaultValue="false">
+          <Annotation Term="Core.Description" String="Entities can be upserted" />
+        </Property>
+        <Property Name="CollectionUpdate" Type="Capabilities.CollectionUpdateType" Nullable="false">
```
Comment (mikepizzo) [inline vocabularies/Org.OData.Capabilities.V1.xml]: >Nullable="false" [](start = 83, length = 16)

Should not be nullable=false

Comment (ralfhandl) [inline vocabularies/Org.OData.Capabilities.V1.xml]: Should be Nullable=true - check other complex properties as well

File: vocabularies/Org.OData.Capabilities.V1.xml (hunk 2 above)
Comment (ralfhandl) [reply vocabularies/Org.OData.Capabilities.V1.xml]: No, we don't need it

File: vocabularies/Org.OData.Capabilities.V1.xml (hunk 4 above)
Comment (ralfhandl) [reply vocabularies/Org.OData.Capabilities.V1.xml]: Should be nullable

File: vocabularies/Org.OData.Capabilities.V1.xml (hunk 3 above)
Comment (ralfhandl) [reply vocabularies/Org.OData.Capabilities.V1.xml]: No, is PATCH on collection using delta format - with differentiation "only upsert" and "upsert and delete"

File: vocabularies/Org.OData.Capabilities.V1.xml [hunk 5]
```diff
@@ -678,6 +713,24 @@ The absence of `RestrictedProperties` denotes all properties are accessible usin
           <Annotation Term="Core.IsLanguageDependent" />
         </Property>
       </ComplexType>
+      <ComplexType Name="CollectionUpdateType">
```
Comment (ralfhandl) [inline vocabularies/Org.OData.Capabilities.V1.xml]: Flatten this, move these four properties to UpdateRestrictionsType, mention in Description that they only are valid if applied to a collection

File: vocabularies/Org.OData.Capabilities.V1.xml [hunk 6]
```diff
@@ -678,6 +713,24 @@ The absence of `RestrictedProperties` denotes all properties are accessible usin
           <Annotation Term="Core.IsLanguageDependent" />
         </Property>
       </ComplexType>
+      <ComplexType Name="CollectionUpdateType">
+        <Property Name="DeltaUpsertable" Type="Edm.Boolean" Nullable="false" DefaultValue="false">
```
Comment (ralfhandl) [inline vocabularies/Org.OData.Capabilities.V1.xml]: Just call this DeltaUpdateSupported

File: vocabularies/Org.OData.Capabilities.V1.xml [hunk 7]
```diff
@@ -678,6 +713,24 @@ The absence of `RestrictedProperties` denotes all properties are accessible usin
           <Annotation Term="Core.IsLanguageDependent" />
         </Property>
       </ComplexType>
+      <ComplexType Name="CollectionUpdateType">
+        <Property Name="DeltaUpsertable" Type="Edm.Boolean" Nullable="false" DefaultValue="false">
+          <Annotation Term="Core.Description"
+            String="Entities can be inserted and updated via a PATCH request with a delta payload" />
+        </Property>
+        <Property Name="DeltaDeletable" Type="Edm.Boolean" Nullable="false" DefaultValue="true">
```
Comment (ralfhandl) [inline vocabularies/Org.OData.Capabilities.V1.xml]: Remove this

File: vocabularies/Org.OData.Capabilities.V1.xml (hunk 4 above)
Comment (ralfhandl) [reply vocabularies/Org.OData.Capabilities.V1.xml]: Removed and moved properties of CollectionUpdateType here

**Pull request merged**